app/controllers/carto/api/templates_controller.rb

```
@@ -0,0 +1,91 @@
+#encoding: UTF-8
+
+module Carto
+  module Api
+    class TemplatesController < ::Api::ApplicationController
+
+      ssl_required :index, :show, :create, :update, :destroy
+      before_filter :load_template, only: [ :show, :update, :destroy ]
+      before_filter :check_feature_flag
+
+      def index
+        templates = Carto::Template.where(organization_id: current_user.organization_id).order(:created_at).reverse
+
+        render_jsonp({ items: templates.map { |template| Carto::Api::TemplatePresenter.new(template).public_values } })
+      rescue => e
+        render json: { error: [e.message] }, status: 400
+      end
+
+      def show
+        render_jsonp({ :errors => ["Template #{params[:id]} not found"] }, 404) and return if @template.nil?
```

Since you can configure which actions do load the template you could return this error in the `before_filter`, removing this simple checks from controller actions ([example](https://github.com/CartoDB/cartodb/blob/master/app/controllers/carto/api/visualizations_controller.rb#L143)).